### PR TITLE
Specify clevis-pkcs11 package for install

### PR DESCRIPTION
**Pull request by Koncpa**
Also adjust phase where is installing clevis
from upstream to install it just in case when
pkc11 package is not available and add conditional for test phase which use official package where with full development of pkcs11.

### CODE REVIEW

File: Sanity/pkcs11/basic/runtest.sh
Comment (kokesak) [inline Sanity/pkcs11/basic/runtest.sh:75]: Maybe you meant to have `-eq 0`  or "unset envvar". In the official released package this implementation is not there, yet I believe. I also think it is not present in the "developer" build either

**Comment by sarroutbi** [comment]
@Koncpa : FYI: clevis-pin-pkcs11 should be available in Fedora 41, Fedora 40, RHEL10 and RHEL9

**Comment by kokesak** [comment]
The tests failed because they do not have installed `jose` package that is being required during the installation.

```
Found CMake: /usr/bin/cmake (3.26.5)
Run-time dependency jansson found: NO (tried pkgconfig and cmake)
Run-time dependency jose found: NO (tried pkgconfig and cmake)

../meson.build:43:7: ERROR: Dependency "jose" not found, tried pkgconfig and cmake
```

The installation through meson should be removed in https://github.com/RedHat-SP-Security/clevis-tests/pull/19. I suggest first finishing and merging this PR and then continue working on this

**Comment by Koncpa** [comment]
> The tests failed because they do not have installed `jose` package that is being required during the installation.
> 
> ```
> Found CMake: /usr/bin/cmake (3.26.5)
> Run-time dependency jansson found: NO (tried pkgconfig and cmake)
> Run-time dependency jose found: NO (tried pkgconfig and cmake)
> 
> ../meson.build:43:7: ERROR: Dependency "jose" not found, tried pkgconfig and cmake
> ```
> 
> The installation through meson should be removed in #19. I suggest first finishing and merging this PR and then continue working on this

@kokesak Do you think that same root cause is for `/Sanity/pkcs11/single-encrypted-disk` for Fedora-40??

**Comment by Koncpa** [comment]
Now it's LGTM. WDYT? @sarroutbi @kokesak 
We could merge this PR, there is prepared conditional for upstream installation of clevis-pkcs11 if rpm missing and PR from @sarroutbi could be dropped. Also CI results looks good.

File: TestHelpers/utils.sh
Comment (kokesak) [inline TestHelpers/utils.sh:24]: I suggest adding info about the installed package. In the logs there is nothing, only a message package is installed.
```
:: [ 16:44:21 ] :: [   INFO   ] :: Package is already installed on system!
```
I would add the outptut of the `rpm -q clevis-pin-pkcs11`

**Comment by kokesak** [comment]
> Now it's LGTM. WDYT? @sarroutbi @kokesak We could merge this PR, there is prepared conditional for upstream installation of clevis-pkcs11 if rpm missing and PR from @sarroutbi could be dropped. Also CI results looks good.

I do not want to block this PR on this, but just looking at the code I think we could call the `development_clevis()` function much earlier. It is now being called a little too late - https://github.com/RedHat-SP-Security/clevis-tests/blob/master/Sanity/pkcs11/basic/runtest.sh#L58. I would put it to the beginning of the setup phase

**Comment by sarroutbi** [comment]
> > Now it's LGTM. WDYT? @sarroutbi @kokesak We could merge this PR, there is prepared conditional for upstream installation of clevis-pkcs11 if rpm missing and PR from @sarroutbi could be dropped. Also CI results looks good.
> 
> I do not want to block this PR on this, but just looking at the code I think we could call the `development_clevis()` function much earlier. It is now being called a little too late - https://github.com/RedHat-SP-Security/clevis-tests/blob/master/Sanity/pkcs11/basic/runtest.sh#L58. I would put it to the beginning of the setup phase

I am not sure this change about setting the call to `development_clevis()` is completely related to this. IMHO, I would:
- Keep this PR as is, including rest of comments.
- Open an issue to clarify why this function should be called before and, if so, fix it in another PR

**Comment by Koncpa** [comment]
I provided requested changes, so let's wait for PR CI and we could merge it. @sarroutbi @kokesak Please feel free to open issue regarding that `development_clevis()` function order.

**Comment by sarroutbi** [comment]
@kokesak : May I ask you to open the issue? I am not sure about the reason why this `development_clevis()` should be called before compared to where it is being done now

**Comment by kokesak** [comment]
> @kokesak : May I ask you to open the issue? I am not sure about the reason why this `development_clevis()` should be called before compared to where it is being done now

Ok, no problem. I added it here - https://github.com/RedHat-SP-Security/clevis-tests/issues/27

**Comment by sarroutbi** [comment]
> > @kokesak : May I ask you to open the issue? I am not sure about the reason why this `development_clevis()` should be called before compared to where it is being done now
> 
> Ok, no problem. I added it here - #27

Great. Thank you very much

**Review by kokesak — APPROVED** [review]
LGTM

**Comment by kokesak** [comment]
I think we can merge this, the centos-stream-10 fail I'm not sure what is about, but it seems that it is TestingFarm related

**Pull request merged**
